### DISCUSSION TITLE
Tests with Moto3 passing locally, but failing in pipeline, any idea why?

### DISCUSSION THREAD

**Discussion by netzo92** [spulec/moto#7341] [General]
`An error occurred (InvalidAccessKeyId) when calling the ListObjects operation: The AWS Access Key Id you provided does not exist in our records.`

This is the error I'm recieving in our gitlab py_test pipeline stage, but locally it works.

Any ideas why this may be happening?

**Comment by bblommers** [comment]
If you see that error, it usually means that the mock isn't working correctly and the tests are reaching out to AWS instead.
See the documentation here: http://docs.getmoto.org/en/latest/docs/getting_started.html#what-about-those-pesky-imports

When you say `locally it works`, do you mean that the tests pass locally? Or does the test run against AWS locally as well, and you just happen to have the correct credentials?

Do you run the full test suite locally, or just some tests?

Is the Gitlab repo open source by any chance? That would make it easier to figure out what's wrong.
